torch/lib/c10d/TCPStore.cpp

```diff
@@ -67,12 +89,30 @@ void TCPStoreDaemon::run() {
       fds[i].revents = 0;
     }
 
+#ifdef _WIN32
+    int res;
+    SYSCHECK_ERR_RETURN_NEG1(
+      res = WSAPoll(fds.data(), fds.size(), checkTimeout_.count()))
+    if (res == 0) {
+      auto rv = WaitForSingleObject(ghStopEvent_, 0);
+      if (rv != WAIT_TIMEOUT) {
+          finished = true;
+          break;
+      }
+      continue;
+    }
+#else
     SYSCHECK_ERR_RETURN_NEG1(::poll(fds.data(), fds.size(), -1));
+#endif
```

Comment:
Looks like it's not that easy to get rid of these ifdefs due to its dependency on vars like `ghStopEvent_`. If that is the case, does it work if we just have one ifdef block in the beginning of this `TCPStoreDaemon::run()` and within that ifdef block we define multiple lambdas for win32 and non-win32 cases? Sth like 

```c++
#ifdef _WIN32
prepare_fd = [&storeListenSocket_]() {...}
#else
prepare_fd = [&storeListenSocket_, & controlPipeFd_]() {...}
#endif
```

@malfet @lw @gunandrose4u please comment if there are better ways. I guess the goal here is clear, which is to have fewer inline ifdefs so that the code can be more readable.